server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -943,4 +943,12 @@ private boolean useConcurrentSearch(Executor concurrentSearchExecutor) {
             return false;
         }
     }
+
+    @Override
+    public int getTargetMaxSliceCount() {
+        if (!isConcurrentSegmentSearchEnabled()) {
+            throw new IllegalStateException("Target slice count should not be used when concurrent search is disabled");
+        }
```

Comment:
Nitpick...the prevailing style is `isConcurrentSegmentSearchEnabled() == false` instead of negation.